packages/devextreme/testing/tests/DevExpress.core/utils.object.tests.js

```diff
@@ -171,7 +171,21 @@ QUnit.test('deepExtendArraySafe utility does not throw an error with \'null\' de
     assert.equal(result.deepProp.toChange, 'changed value');
 });
 
-QUnit.test('deepExtendArraySafe utility does not pollute object prototype', function(assert) {
-    objectUtils.deepExtendArraySafe({ }, JSON.parse('{ "__proto__": { "pollution": true }}'), true);
-    assert.ok(!('pollution' in { }), 'object prototype is not polluted');
+QUnit.test('deepExtendArraySafe sets undefined', function(assert) {
+    const objWithValue = { time: { duration: 50 } };
+    const objNoValue = {};
+    objectUtils.deepExtendArraySafe(objWithValue, { time: { duration: undefined } }, true);
+    objectUtils.deepExtendArraySafe(objNoValue, { time: { duration: undefined } }, true);
+
+    assert.equal(objWithValue.time.duration, 50);
+    assert.notOk(Object.prototype.hasOwnProperty.call(objNoValue.time, 'duration'));
+
+    objectUtils.deepExtendArraySafe(objWithValue, { time: { duration: undefined } }, true, false, true);
+    objectUtils.deepExtendArraySafe(objNoValue, { time: { duration: undefined } }, true, false, true);
+
+    assert.equal(objWithValue.time.duration, undefined);
```

Comment:
Let's also check that the ```time``` object has own property ```duration``` here
Because if in runtime we get just an empty time object (e.g ```time: {}```),
the ```time. duration``` value will be ```undefined``` too